crates/event-streaming/build.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fs, path::PathBuf, process::Command};
+
+const SUI_REPO_URL: &str = "https://github.com/MystenLabs/sui.git";
+const SUI_TAG: &str = "testnet-v1.50.1";
+const GOOGLEAPIS_REPO_URL: &str = "https://github.com/googleapis/googleapis.git";
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let sui_proto_source = ensure_sui_proto_source()?;
+    let googleapis_source = ensure_googleapis_source()?;
+
+    let proto_files = vec!["proto/walrus/event/v1/service.proto"];
+
+    // Check that our proto files exist
+    for proto_file in &proto_files {
+        if !PathBuf::from(proto_file).exists() {
+            return Err(format!("Proto file not found: {}", proto_file).into());
+        }
+    }
+
+    tonic_build::configure()
+        .build_server(true)
+        .build_client(true)
+        .include_file("mod.rs")
+        .compile_protos(
+            &proto_files,
+            &[
+                "proto",
+                &sui_proto_source.to_string_lossy(),
+                &googleapis_source.to_string_lossy(),
+            ],
+        )?;
+
+    // Tell cargo to rerun if proto files change
+    println!("cargo:rerun-if-changed=proto/");
+    println!("cargo:rerun-if-changed={}", googleapis_source.display());
+
+    // Also rerun if Sui proto files change (if they exist)
+    if sui_proto_source.exists() {
+        println!("cargo:rerun-if-changed={}", sui_proto_source.display());
+    }
+
+    Ok(())
+}
+
+fn ensure_sui_proto_source() -> Result<PathBuf, Box<dyn std::error::Error>> {
+    // Priority order for finding Sui proto files:
+    // 1. Environment variable SUI_REPO_PATH
+    // 2. Relative paths (assuming walrus and sui are siblings)
+    // 3. Download from GitHub to build cache
+
+    if let Ok(sui_path) = std::env::var("SUI_REPO_PATH") {
+        let sui_proto_path = PathBuf::from(sui_path).join("crates/sui-rpc-api/proto");
+        if sui_proto_path.exists() {
+            println!(
```

Comment:
What guarantees that the sui repo in this case is checked out to the correct SHA?